src/display/lv_display.c

```diff
@@ -405,7 +405,11 @@ void lv_display_set_buffers(lv_display_t * disp, void * buf1, void * buf2, uint3
 
     lv_color_format_t cf = lv_display_get_color_format(disp);
     uint32_t stride = lv_draw_buf_width_to_stride(w, cf);
-    if(stride * h > buf_size && render_mode != LV_DISPLAY_RENDER_MODE_PARTIAL) {
+    if(render_mode == LV_DISPLAY_RENDER_MODE_PARTIAL) {
+        h = buf_size / stride;
+        LV_ASSERT_MSG(h != 0, "the buffer is too small for PARTIAL mode");
+    }
```

Comment:
Add `return` here in case of asserts are disabled?

Reply:
I added return before, but @FASTSHIFT asked me to remove it...

Reply:
Added return ...again...

Reply:
I think below is more suitable.
```c
h = buf_size / stride;
LV_ASSERT_MSG(h > 0, "the buffer is too small for PARTIAL mode");
return;
```

Reply:
> I think below is more suitable.
> 
> ```c
> h = buf_size / stride;
> LV_ASSERT_MSG(h > 0, "the buffer is too small for PARTIAL mode");
> return;
> ```

well, I think in this case it will always `return`

Reply:
> well, I think in this case it will always return

Wow, you are right! This?
```c
    if(render_mode == LV_DISPLAY_RENDER_MODE_PARTIAL) {
        h = buf_size / stride;
    }
    ...
    if(h == 0) {
       LV_ASSERT_MSG(h != 0, "the buffer is too small");
       return; 
    }   

```

Reply:
We calculate the h for partial mode based on the stride and buf size.

Reply:
I wonder whether `return;` is still needed after `LV_ASSERT`, because in current configuration `LV_ASSERT()` and `LV_ASSERT_MSG()` macros end up in infinite loop:
https://github.com/lvgl/lvgl/blob/master/src/misc/lv_assert.h#L37
https://github.com/lvgl/lvgl/blob/master/lv_conf_template.h#L232

(but correct me if I'm wrong)

Reply:
The ASSERTS can be disabled. In this case they do nothing but we still should return on error.